### PR TITLE
ci: migration gha pr prerelease

### PR DESCRIPTION
**Pull request by jportasa**
- [x] Add version print logic into codebase 
- [x] Add docker login for pull rate limitations 
- [x] Add snyk step as `needs` for prerelease once it will be ok
- [x] Migrate into Org secrets
- [x] Change upgrade codes for windows (keep old codes)
- [x] Squash commits because Joan Porta did not accept CLA

### CODE REVIEW

File: .github/workflows/prerelease.yml [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+          password: ${{ secrets.OHAI_DOCKER_HUB_PASSWORD }}
+      - name: Pre release
+        run: make ci/prerelease
+      - name: Notify failure via Slack
+        if: ${{ failure() }}
+        uses: archive/github-actions-slack@master
+        with:
+          slack-bot-user-oauth-access-token: ${{ secrets.OHAI_SLACK_TOKEN }}
+          slack-channel: ${{ env.OHAI_SLACK_CHANNEL }}
+          slack-text: "Repo: ${{ env.REPO_FULL_NAME }}, prerelease for nix FAILURE!"
```
Comment (roobre) [inline .github/workflows/prerelease.yml:106]:
```suggestion
          slack-text: "❌ `${{ env.REPO_FULL_NAME }}`: prerelease pipeline failed."
```

File: .github/workflows/prerelease.yml [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+        shell: bash
+        run: |
+          build/windows/upload_msi.sh ${INTEGRATION} ${{ matrix.goarch }} ${TAG}
+      - name: Notify failure via Slack
+        if: ${{ failure() }}
+        uses: archive/github-actions-slack@master
+        with:
+          slack-bot-user-oauth-access-token: ${{ secrets.OHAI_SLACK_TOKEN }}
+          slack-channel: ${{ env.OHAI_SLACK_CHANNEL }}
+          slack-text: "Repo: ${{ env.REPO_FULL_NAME }}, prerelease for Windows FAILURE!"
```
Comment (roobre) [inline .github/workflows/prerelease.yml:149]:
```suggestion
          slack-text: "❌ `${{ env.REPO_FULL_NAME }}`: prerelease pipeline failed."
```

File: build/windows/upload_msi.sh [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+#
+#
+# Upload msi artifacts to GH Release assets
+#
+#
+INTEGRATION=$1
+ARCH=$2
+TAG=$3
+
+hub release edit -a "build/package/windows/nri-${ARCH}-installer/bin/Release/nri-${INTEGRATION}-${ARCH}.${TAG:1}.msi" -m ${TAG} ${TAG}
```
Comment (roobre) [inline build/windows/upload_msi.sh:12]: Try to replace `hub` with `gh` if possible: https://github.com/newrelic/nri-mssql/commit/c126b3815244f309eecc60d0fea94eb9b40adf4d.patch

**Review by roobre — CHANGES_REQUESTED** [review]

File: .github/workflows/prerelease.yml [hunk 4]
```diff
@@ -0,0 +1,149 @@
...
+          username: ${{ secrets.OHAI_DOCKER_HUB_ID }}
+          password: ${{ secrets.OHAI_DOCKER_HUB_PASSWORD }}
+      - name: Pre release
+        run: make ci/prerelease
+      - name: Notify failure via Slack
+        if: ${{ failure() }}
+        uses: archive/github-actions-slack@master
+        with:
+          slack-bot-user-oauth-access-token: ${{ secrets.OHAI_SLACK_TOKEN }}
+          slack-channel: ${{ env.OHAI_SLACK_CHANNEL }}
```
Comment (roobre) [inline .github/workflows/prerelease.yml:105]:
```suggestion
          slack-bot-user-oauth-access-token: ${{ secrets.COREINT_SLACK_TOKEN }}
          slack-channel: ${{ env.COREINT_SLACK_CHANNEL }}
```

File: .github/workflows/prerelease.yml [hunk 5]
```diff
@@ -0,0 +1,149 @@
...
+      - name: Upload MSI to GH
+        shell: bash
+        run: |
+          build/windows/upload_msi.sh ${INTEGRATION} ${{ matrix.goarch }} ${TAG}
+      - name: Notify failure via Slack
+        if: ${{ failure() }}
+        uses: archive/github-actions-slack@master
+        with:
+          slack-bot-user-oauth-access-token: ${{ secrets.OHAI_SLACK_TOKEN }}
+          slack-channel: ${{ env.OHAI_SLACK_CHANNEL }}
```
Comment (roobre) [inline .github/workflows/prerelease.yml:148]:
```suggestion
          slack-bot-user-oauth-access-token: ${{ secrets.COREINT_SLACK_TOKEN }}
          slack-channel: ${{ env.COREINT_SLACK_CHANNEL }}
```

**Review by roobre — CHANGES_REQUESTED** [review]
I missed this earlier, sorry :(

File: build/package/windows/nri-386-installer/Product.wxs [hunk 6]
```diff
@@ -1,11 +1,11 @@
 ﻿<?xml version="1.0" encoding="UTF-8"?>
 <Wix xmlns="http://schemas.microsoft.com/wix/2006/wi">
-    <?define Version = "{IntegrationVersion}" ?>
+    <?define Version = "$(var.IntegrationVersion)" ?>
     <?define ProductId = "*" ?>
-    <?define UpgradeCode = "{59bf8deb-66d6-4a6f-a520-6cd60de9c399}" ?>
+    <?define UpgradeCode = "{c7d10ef9-3430-4195-8d46-52548dd4ad07}" ?>
```
Comment (paologallinaharbur) [inline build/package/windows/nri-386-installer/Product.wxs:5]: This should not change

File: build/package/windows/nri-amd64-installer/Product.wxs [hunk 7]
```diff
@@ -1,11 +1,11 @@
 ﻿<?xml version="1.0" encoding="UTF-8"?>
 <Wix xmlns="http://schemas.microsoft.com/wix/2006/wi">
-    <?define Version = "{IntegrationVersion}" ?>
+    <?define Version = "$(var.IntegrationVersion)" ?>
     <?define ProductId = "*" ?>
-    <?define UpgradeCode = "{59bf8deb-66d6-4a6f-a520-6cd60de9c399}" ?>
+    <?define UpgradeCode = "{4d3f0446-564f-11eb-ae93-0242ac130002}" ?>
```
Comment (paologallinaharbur) [inline build/package/windows/nri-amd64-installer/Product.wxs:5]: Also this should not change

File: build/package/windows/nri-386-installer/Product.wxs (hunk 6 above)
Comment (paologallinaharbur) [inline build/package/windows/nri-386-installer/Product.wxs:5]:
```suggestion
    <?define UpgradeCode = "{59bf8deb-66d6-4a6f-a520-6cd60de9c399}" ?>
```

File: build/package/windows/nri-amd64-installer/Product.wxs (hunk 7 above)
Comment (paologallinaharbur) [reply build/package/windows/nri-amd64-installer/Product.wxs:5]:
```suggestion
    <?define UpgradeCode = "{59bf8deb-66d6-4a6f-a520-6cd60de9c399}" ?>
```

**Comment by paologallinaharbur** [comment]
Fixed the slack secret naming and the upgrade codes

File: .github/workflows/prerelease.yml [hunk 8]
```diff
@@ -0,0 +1,149 @@
...
+          username: ${{ secrets.OHAI_DOCKER_HUB_ID }}
+          password: ${{ secrets.OHAI_DOCKER_HUB_PASSWORD }}
+      - name: Pre release
+        run: make ci/prerelease
+      - name: Notify failure via Slack
+        if: ${{ failure() }}
+        uses: archive/github-actions-slack@master
+        with:
+          slack-bot-user-oauth-access-token: ${{ secrets.COREINT_SLACK_TOKEN }}
+          slack-channel: ${{ env.COREINT_SLACK_CHANNEL }}
```
Comment (paologallinaharbur) [inline .github/workflows/prerelease.yml:105]:
```suggestion
          slack-channel: ${{ secrets.COREINT_SLACK_CHANNEL }}
```

File: .github/workflows/prerelease.yml [hunk 9]
```diff
@@ -0,0 +1,149 @@
...
+      - name: Upload MSI to GH
+        shell: bash
+        run: |
+          build/windows/upload_msi.sh ${INTEGRATION} ${{ matrix.goarch }} ${TAG}
+      - name: Notify failure via Slack
+        if: ${{ failure() }}
+        uses: archive/github-actions-slack@master
+        with:
+          slack-bot-user-oauth-access-token: ${{ secrets.COREINT_SLACK_TOKEN }}
+          slack-channel: ${{ env.COREINT_SLACK_CHANNEL }}
```
Comment (paologallinaharbur) [inline .github/workflows/prerelease.yml:148]:
```suggestion
          slack-channel: ${{ secrets.COREINT_SLACK_CHANNEL }}
```

File: .github/workflows/prerelease.yml [hunk 10]
```diff
@@ -0,0 +1,149 @@
...
+    steps:
+      - uses: actions/checkout@v2
+      - name: Login to DockerHub
+        uses: docker/login-action@v1
+        with:
+          username: ${{ secrets.OHAI_DOCKER_HUB_ID }}
+          password: ${{ secrets.OHAI_DOCKER_HUB_PASSWORD }}
+      - name: Scan code for vulnerabilities
+        env:
+          SNYK_TOKEN: ${{ secrets.COREINT_SNYK_TOKEN }}
```
Comment (roobre) [inline .github/workflows/prerelease.yml:45]: Not sure if it is legal to put `env` in `step` context, but we were usually putting it in the `job` just after `runs-on`.

**Review by roobre — CHANGES_REQUESTED** [review]

File: .github/workflows/prerelease.yml (hunk 10 above)
Comment (paologallinaharbur) [reply .github/workflows/prerelease.yml:45]: yes, I'll move it

**Review by roobre — APPROVED** [review]

**Review by paologallinaharbur — APPROVED** [review]

**Pull request merged**